### PR TITLE
Warn when OrbaxCheckpoint fails to finalize checkpoints

### LINKED ISSUE #23609: OrbaxCheckpoint silently discards errors when finalizing checkpoints
`OrbaxCheckpoint` discards any error raised while finalizing checkpoints, so a run whose last checkpoint never landed finishes exactly like one that saved correctly.

`on_train_end`, in `keras/src/callbacks/orbax_checkpoint.py`:

```python
def on_train_end(self, logs=None):
    # Close the Checkpointer - this waits for any pending async saves
    # to complete before closing
    try:
        self.checkpointer.close()
    except Exception:
        pass  # Ignore errors during cleanup
```

Closing is what surfaces failures from pending saves, so this is the one place a failed write can report itself. With `save_on_background=True` the save is a future, and if it never completes, `close()` raises and that exception is dropped.

I hit this on CI. The async save timed out:

```
ChainedFuture completed 0/1 futures but timed out after 1199.96 seconds
Skipping merge of OCDBT checkpoints: No per-process OCDBT checkpoint subdirs found in /tmp/...
```

`fit()` returned normally. The failure only showed up much later, as restored weights that were one optimizer step stale:

```
Mismatched elements: 80 / 80 (100%)
Max absolute difference among violations: 0.00373325
```

Reproducing it is simple: make `close()` raise and nothing is reported.

```python
callback = OrbaxCheckpoint(directory=d, save_freq="epoch")
model.fit(x, y, epochs=1, verbose=0, callbacks=[callback])

class Failing:
    def __init__(self, inner): self._inner = inner
    def __getattr__(self, name): return getattr(self._inner, name)
    def close(self): raise TimeoutError("pending save did not complete")

callback.checkpointer = Failing(callback.checkpointer)
callback.on_train_end()   # no warning, no error
```

At minimum this should warn with the underlying error, so a lost checkpoint is distinguishable from a saved one. Happy to send the PR.

### PR DESCRIPTION
**Pull request by GargiGupta-io**
## Description

Fixes #23609.

`on_train_end` closes the checkpointer inside `except Exception: pass`. Closing is what surfaces failures from pending saves, so a run whose last checkpoint never landed finished exactly like one that saved correctly — the loss only showed up later as restored weights that were a step stale.

This warns with the underlying error instead of discarding it. I kept it a warning rather than a re-raise so that genuinely benign cleanup errors don't start failing runs at the very end of training; happy to make it raise if you'd prefer.

## Contributor Agreement

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

### CODE REVIEW

**Review by hertschuh — APPROVED** [review]
Thanks for the improvement!
